src/core/Form/TextInput/TextInput.tsx

```diff
@@ -1,74 +1,117 @@
-import React, { Component } from 'react';
+import React, { Component, ReactNode, ChangeEvent, FocusEvent } from 'react';
 import { default as styled } from 'styled-components';
 import { withSuomifiDefaultProps } from '../../theme/utils';
+import {
+  HtmlInput,
+  HtmlInputProps,
+  HtmlDiv,
+  HtmlDivProps,
+} from '../../../reset';
 import { TokensProp, InternalTokensProp } from '../../theme';
 import { baseStyles } from './TextInput.baseStyles';
-import {
-  TextInput as CompTextInput,
-  TextInputProps as CompTextInputProps,
-} from '../../../components/Form/TextInput';
+import { LabelText, LabelMode } from '../LabelText/LabelText';
+import { StatusText } from '../StatusText/StatusText';
+import { InputStatus } from '../types';
+import { HintText } from '../HintText/HintText';
 import classnames from 'classnames';
 import { Icon, IconProps, BaseIconKeys } from '../../Icon/Icon';
 import { Omit } from '../../../utils/typescript';
+import { idGenerator } from '../../../utils/uuid';
 
 const baseClassName = 'fi-text-input';
 export const textInputClassNames = {
   baseClassName,
-  labelParagraph: `${baseClassName}_label-p`,
+  fullWidth: `${baseClassName}--full-width`,
+  disabled: `${baseClassName}--disabled`,
   error: `${baseClassName}--error`,
   success: `${baseClassName}--success`,
   icon: `${baseClassName}_with-icon`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+  functionalityContainer: `${baseClassName}_functionality-container`,
 };
 
-export interface TextInputProps extends CompTextInputProps, TokensProp {
+export interface TextInputProps
+  extends Omit<HtmlInputProps, 'type'>,
+    TokensProp {
+  /** TextInput container div class name for custom styling. */
+  className?: string;
+  /** TextInput container div props */
+  inputContainerProps?: Omit<HtmlDivProps, 'className'>;
+  /** Disable input usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on input text change */
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  /** To execute on input text onBlur */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** A custom element to be passed to the component. Will be rendered after the input */
```

Comment:
What's the use case for the children prop nowadays? I think it was used for the search icon in the old structure, but what are the other ones?

Reply:
Good question.
I have wondered the same in the past, but can't remember that there was any other specific use case.

I would suggest that we can get rid of this prop also.

Reply:
I guess it's good for general flexibility. Maybe for tooltips or such, but not sure if something like that needs to be supported inside the component. Might be something we can get rid of, but let's think it through well before doing anything.

Reply:
I propose we remove it. We are moving away from unstyled versions anyways, so why support something that probably break the layout? And there will be breaking changes to the API anyways, so now would be a good time in my opinion.

Reply:
`children` prop is now removed. Updated the PR description accordingly.